src/main/java/com/moneylendingapp/security/jwt/JwtUtil.java

```diff
@@ -0,0 +1,62 @@
+package com.moneylendingapp.security.jwt;
+
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.SignatureAlgorithm;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.stereotype.Service;
+
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class JwtUtil {
+
+    private final JwtConfig jwtConfig;
+
+    public String extractUsername(String token){
+        return extractClaim(token, Claims::getSubject);
+    }
+
+    public Date extractExpiration(String token){
+        return extractClaim(token, Claims::getExpiration);
+    }
+
+    public <T> T extractClaim(String token, Function<Claims, T> claimsResolver){
+        final Claims claims = extractAllClaims(token);
+        return claimsResolver.apply(claims);
+    }
+
+    public Claims extractAllClaims(String token){
+        return Jwts.parser().setSigningKey(jwtConfig.getSECRET_KEY()).parseClaimsJws(token).getBody();
+    }
+
+    private Boolean isTokenExpired(String token){
+        return extractExpiration(token).before(new Date());
```

Comment:
Might be better to use Clock and Local Date in this class

Reply:
Saw that the before method only takes Date